bot/cogs/games/ttt.py

```diff
@@ -0,0 +1,237 @@
+        await self.bot.add_reaction(msg, u"\u2196")  # 0 tl
+        await self.bot.add_reaction(msg, u"\u2B06")  # 1 t
+        await self.bot.add_reaction(msg, u"\u2197")  # 2 tr
+        await self.bot.add_reaction(msg, u"\u2B05")  # 3 l
+        await self.bot.add_reaction(msg, u"\u23FA")  # 4 mid
+        await self.bot.add_reaction(msg, u"\u27A1")  # 5 r
+        await self.bot.add_reaction(msg, u"\u2199")  # 6 bl
+        await self.bot.add_reaction(msg, u"\u2B07")  # 7 b
+        await self.bot.add_reaction(msg, u"\u2198")  # 8 br
+
+    async def on_reaction_add(self, reaction: Reaction, user: User) -> None:
+        if reaction.message.author.id == self.bot.user.id and not user.id == self.bot.user.id:
+            move = self.decodeMove(str(reaction.emoji))
+            if move is not None:
+                await self.ttt_move(user, reaction.message, move)
+
+    def decodeMove(self, emoji: str) -> None:
+        emoji_dict = {
+            u"\u2196": 0,
+            u"\u2B06": 1,
+            u"\u2197": 2,
+            u"\u2B05": 3,
+            u"\u23FA": 4,
+            u"\u27A1": 5,
+            u"\u2199": 6,
+            u"\u2B07": 7,
+            u"\u2198": 8
+        }
+        return emoji_dict[emoji] if emoji in emoji_dict else None
+
+    # Utility Functions
+    def tttTable(self, xo) -> str:
+        return (("%s%s%s\n" * 3) % tuple(xo)).replace("o", ":o2:").replace("x", ":regional_indicator_x:").replace(" ", ":white_large_square:")
+
+    def tttMatrix(self, b: list) -> list:
+        return [
+            [b[0], b[1], b[2]],
+            [b[3], b[4], b[5]],
+            [b[6], b[7], b[8]]
+        ]
+
+    def tttCoordsToIndex(self, coords: tuple) -> str:
+        board_map = {
+            (0, 0): 0,
+            (0, 1): 1,
+            (0, 2): 2,
+            (1, 0): 3,
+            (1, 1): 4,
+            (1, 2): 5,
+            (2, 0): 6,
+            (2, 1): 7,
+            (2, 2): 8
+        }
+        return board_map[coords]
+
+    def tttDoChecks(self, b: str) -> t.Union[str, None]:
+        m = self.tttMatrix(b)
+        if self.tttCheckWin(m, "x"):
+            return "win X"
+        if self.tttCheckWin(m, "o"):
+            return "win O"
+        if self.tttCheckDraw(b):
+            return "draw"
+        return None
+
+    def tttFindStreaks(self, m: list, xo: str) -> tuple:
+        row = [0, 0, 0]
+        col = [0, 0, 0]
+        dia = [0, 0]
+
+        # Check rows and columns for X streaks
+        for y in range(3):
+            for x in range(3):
+                if m[y][x] == xo:
+                    row[y] += 1
+                    col[x] += 1
+
+        # Check diagonals
+        if m[0][0] == xo:
+            dia[0] += 1
+        if m[1][1] == xo:
+            dia[0] += 1
+            dia[1] += 1
+        if m[2][2] == xo:
+            dia[0] += 1
+        if m[2][0] == xo:
+            dia[1] += 1
+        if m[0][2] == xo:
+            dia[1] += 1
+
+        return (row, col, dia)
```

Comment:
sure

Reply:
but i have a question,
```py
def ttt_ai_think(self, m: list) -> t.Union[tuple, bool]:
        rx, cx, dx = self.ttt_find_streaks(m, "x")
        ro, co, do = self.ttt_find_streaks(m, "o")

        mv = self.ttt_ai_move(2, m, ro, co, do)
        if mv is not False:
            return mv
        mv = self.ttt_ai_move(2, m, rx, cx, dx)
        if mv is not False:
            return mv
        mv = self.ttt_ai_move(1, m, ro, co, do)
        if mv is not False:
            return mv
        return self.ttt_ai_move(1, m, rx, cx, dx)
```
this uses the streak function, what should i do, and when the win function returns a tuple, who should be the winner, how do i find that?

Reply:
the readability of this is really bad, also, do we even want AI for tic tac toe here? I can design it in a better way, but if I do it will always either win to draw, it will never lose which may be disappointing for users